File: 3321-type-of-triangle/3321-type-of-triangle.cpp
class Solution {
public:
    string triangleType(vector<int>& nums) {
        set<int> s;
        s.insert(nums[0]);
        s.insert(nums[1]);
        s.insert(nums[2]);
        if(nums[2] >= nums[0]+nums[1] || nums[0] >= nums[2]+nums[1] || nums[1] >= nums[2]+nums[0]){
            return "none";
        }
        if(s.size()==1){
            return "equilateral";
        }
        else if(s.size()==2){
            return "isosceles";
        }
        else{
            return "scalene";
        }
    }
};

File: 3321-type-of-triangle/3321-type-of-triangle.rs
impl Solution {
    pub fn triangle_type(nums: Vec<i32>) -> String {
        if nums[0] + nums[1] <= nums[2] || nums[0] + nums[2] <= nums[1] || nums[2] + nums[1] <= nums[0]{
            return "none".to_string();
        }
        else if nums[0] == nums[1] && nums[1] == nums[2] {
            return "equilateral".to_string();
        }
        else if nums[0] == nums[1] || nums[1] == nums[2] || nums[2] == nums[0] {
            return "isosceles".to_string();
        }
        return "scalene".to_string();
    }
//Learnings:
//1) " " -> always &'static str, so to convert it back to String, we use .to_string()
}